eng/packaging.targets

```diff
@@ -63,71 +61,55 @@
-        <None Include="$(PlaceholderFile)" PackagePath="lib\xamarinmac20" Pack="true" />
-        <None Include="$(PlaceholderFile)" PackagePath="lib\xamarintvos10" Pack="true" />
-        <None Include="$(PlaceholderFile)" PackagePath="lib\xamarinwatchos10" Pack="true" />
+        <None Include="$(PlaceholderFile)" PackagePath="$(BuildOutputTargetFolder)/MonoAndroid10" Pack="true" />
+        <None Include="$(PlaceholderFile)" PackagePath="$(BuildOutputTargetFolder)/MonoTouch10" Pack="true" />
+        <None Include="$(PlaceholderFile)" PackagePath="$(BuildOutputTargetFolder)/xamarinios10" Pack="true" />
+        <None Include="$(PlaceholderFile)" PackagePath="$(BuildOutputTargetFolder)/xamarinmac20" Pack="true" />
+        <None Include="$(PlaceholderFile)" PackagePath="$(BuildOutputTargetFolder)/xamarintvos10" Pack="true" />
+        <None Include="$(PlaceholderFile)" PackagePath="$(BuildOutputTargetFolder)/xamarinwatchos10" Pack="true" />
       </ItemGroup>
     </When>
   </Choose>
 
   <!-- TODO: Remove this target when no library relies on the intellisense documentation file anymore.-->
   <Target Name="ChangeDocumentationFileForPackaging"
           AfterTargets="DocumentationProjectOutputGroup"
-          Condition="'$(ExcludeFromPackage)' != 'true' and
-                     '$(UseIntellisenseDocumentationFile)' == 'true'">
+          Condition="'$(UseIntellisenseDocumentationFile)' == 'true'">
     <ItemGroup>
       <DocumentationProjectOutputGroupOutput Remove="@(DocumentationProjectOutputGroupOutput)" />
       <DocumentationProjectOutputGroupOutput Include="$(LibIntellisenseDocumentationFilePath)" />
     </ItemGroup>
   </Target>
 
+  <!-- Add runtime specific file into the package if the tfm is RID specific or if the tfm targets .NETFramework and netstandard2.0-windows TFM is present.
+       This is necessary as compatible RID specific tfms win over compatible RID agnostic tfms. Hence without this, netstandard2.0-windows would win over net461. -->
   <Target Name="AddRuntimeSpecificFilesToPackage"
           DependsOnTargets="BuiltProjectOutputGroup;
                             DocumentationProjectOutputGroup;
                             SatelliteDllsProjectOutputGroup;
                             $(TargetsForTfmSpecificBuildOutput)"
-          Condition="'$(ExcludeFromPackage)' != 'true' and
-                     '$(TargetsAnyOS)' != 'true'">
+          Condition="'$(TargetFrameworkSuffix)' != '' or
+                     ('$(TargetFrameworkIdentifier)' == '.NETFramework' and $(TargetFrameworks.Contains('netstandard2.0-windows')))">
+    <PropertyGroup>
+      <RuntimeSymbolPath>$(TargetDir)$(TargetName).pdb</RuntimeSymbolPath>
+      <_packageTargetRuntime Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'">$(PackageTargetRuntime)</_packageTargetRuntime>
+      <_packageTargetRuntime Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">win</_packageTargetRuntime>
```

Comment:
What happens when someone actually targets `net4.8-windows` or even does something like `net4.8-unix`?  Might be helpful to have a hard block on that just to prevent people from doing it since it will give the wrong thing for P2Ps as well.

Reply:
NuGet is the hard block here. When using an unknown tfm, NuGet will throw. I plan to update [this pattern](https://github.com/dotnet/arcade/blob/eecf5030126490497f2ff9fe4b7ecd4c546c0f30/src/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk/src/build/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.props#L14) to not strip potential suffixes anymore from .NETFramework tfms after this change is in (and when Arcade branched-off for NET7).

Reply:
Cool.  I think arcade may have branched here: https://github.com/dotnet/arcade/tree/release/6.0

Reply:
https://github.com/dotnet/arcade/pull/7863